V1.5 (#4)
tigergraph/pytigergraph-docs: modules/object_oriented_schema/pages/schema-def.adoc

To add an AccountHolder vertex and a HOLDS_ACCOUNT edge to the Ethereum dataset, simply:

```py
from pyTigerGraph import TigerGraphConnection
from pyTigerGraph.schema import Graph, Vertex, Edge

from datetime import datetime
from typing import List, Dict, Optional, Union
from dataclasses import dataclass, fields

conn = TigerGraphConnection(host="http://YOUR_HOSTNAME_HERE", graphname="Ethereum")

g = Graph(conn)

@dataclass
class AccountHolder(Vertex):
    name: str
    address: str
    accounts: List[str]
    dob: datetime
    some_map: Dict[str, int]
    some_double: "DOUBLE"
    primary_id: str = "name"  # always of type string, corresponds to the desired primary ID attribute.
    primary_id_as_attribute: bool = True

@dataclass
class HOLDS_ACCOUNT(Edge):
    opened_on: datetime
    from_vertex: Union[AccountHolder, g.vertex_types["Account"]]
    to_vertex: g.vertex_types["Account"]
    is_directed: bool = True
    reverse_edge: str = "ACCOUNT_HELD_BY"
    discriminator: str = "opened_on"

g.add_vertex_type(AccountHolder)

g.add_edge_type(HOLDS_ACCOUNT)

g.commit_changes()
```

> One could also define the entire graph schema using the approach. For example, for the Cora dataset, the schema would look something like this:

Reviewer: I'm a little confused here by "One could also define the entire graph schema using the approach." 

Which approach are you referring to? If it is the approach following this statement. Consider:

```suggestion
Users can define an entire graph schema in the approach below. Using the Cora dataset example, the schema would look something like this:
```
